### ISSUE TITLE
`NSURLSession.dataTaskWithRequest` can be monitored, but the request body is not captured

### ISSUE THREAD

**Issue by goldshtn** [Subito-it/SBTUITestTunnel#6]
When using [`NSURLSession.dataTaskWithRequest`](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSession_class/#//apple_ref/doc/uid/TP40013435-CH1-SW15) with a POST request, the request can have an `HTTPBody`. Even though this kind of request can be stubbed and monitored, the resulting `SBTMonitoredNetworkRequest.request` object will never have an `HTTPBody` set. In my UI tests, my request monitoring logic requires access to the `HTTPBody`, so this is a bit of a blocker.

The root cause seems to be that the [`HTTPBodyFix` category on `NSURLSession`](https://github.com/Subito-it/SBTUITestTunnel/blob/master/Pod/Server/NSURLSession%2BHTTPBodyFix.m) doesn't swizzle `dataTaskWithRequest`, but rather only `uploadTaskWithRequest`. To verify, I added the following swizzled method to that category, and was able to successfully retrieve the `HTTPBody` from `SBTMonitoredNetworkRequest.request` in my tests:

```objective-c
- (NSURLSessionDataTask *)swz_dataTaskWithRequest:(NSURLRequest *)request
                                completionHandler:(void (^)(NSData *data,
                                                            NSURLResponse *response,
                                                            NSError *error))completionHandler
{
    if ([request isKindOfClass:[NSMutableURLRequest class]] && request.HTTPBody) {
        [NSURLProtocol setProperty:request.HTTPBody forKey:SBTUITunneledNSURLProtocolHTTPBodyKey inRequest:(NSMutableURLRequest *)request];
    }
    
    return [self swz_dataTaskWithRequest:request completionHandler:completionHandler];
}

// And then in +load:
SBTTestTunnelInstanceSwizzle(self.class, @selector(dataTaskWithRequest:completionHandler:), @selector(swz_dataTaskWithRequest:completionHandler:));
```

There are multiple versions of `dataTaskWithRequest`, and I think they all need to be added to this category to capture the `HTTPBody` property if it was provided. If you'd like, I can try to prepare a PR -- it's basically just swizzling these additional methods that needs to be added, as far as I can tell.

**Comment by tcamin** [comment]
I was previously swizzling uploadTaskWithRequest and downloadTaskWithRequest in a similar fashion as the one in your example (check the file's history). This was however leading to some  crashes that were solved with the current implementation. I'm on mobile now and for a week I won't be able to check this, but IIRC it was a problem with Background NSURLSessions. Could you try to see if your fix works with default and background NSURLSessions?

I hadn't time to reorganize UI Tests for the framework but it would help to have tests covering all these cases.

**Comment by tcamin** [comment]
Should be fixed in 1.0.3

**Comment by goldshtn** [comment]
Looks great, thanks for the quick turnaround.
